File: tests/shots.py
"""Screenshots of the main states. Run: python3 tests/shots.py [light|dark]  -> tests/out/*.png"""
import asyncio, sys
from playwright.async_api import async_playwright
from common import launch, attach_logs, load_sample, OUT, PAGE

async def main():
    scheme = sys.argv[1] if len(sys.argv) > 1 else 'light'
    async with async_playwright() as p:
        b = await launch(p)
        pg = await b.new_page(viewport={'width': 1440, 'height': 900}, color_scheme=scheme)
        logs = []; attach_logs(pg, logs)
        await pg.goto(PAGE)
        await pg.wait_for_timeout(2500)
        await pg.screenshot(path=str(OUT / f'0_landing_{scheme}.png'))
        await load_sample(pg)
        await pg.screenshot(path=str(OUT / f'1_map_{scheme}.png'))
        await pg.click('#viewSeg button[data-v=split]'); await pg.wait_for_timeout(5000)
        await pg.screenshot(path=str(OUT / f'2_split_{scheme}.png'))
        await pg.click('#colorSeg button[data-v=mode]')
        await pg.evaluate("setTimeRange(Date.UTC(2025,8,8), Date.UTC(2025,8,17), 'x')"); await pg.wait_for_timeout(4000)
        await pg.screenshot(path=str(OUT / f'3_filtered_{scheme}.png'))
        await pg.evaluate("setTimeRange(null,null,'x'); map.jumpTo({center:[-9.14,38.73], zoom:11.5})"); await pg.wait_for_timeout(1500)
        await pg.evaluate("S.play.clock = Date.UTC(2025,5,4,7,0); S.play.session = true; drawFrame(true)")
        await pg.click('#playBtn'); await pg.wait_for_timeout(3000); await pg.click('#playBtn'); await pg.wait_for_timeout(2500)
        await pg.screenshot(path=str(OUT / f'4_playback_{scheme}.png'))
        await pg.click('#viewSeg button[data-v=cube]'); await pg.wait_for_timeout(4000)
        await pg.screenshot(path=str(OUT / f'5_cube_{scheme}.png'))
        for l in logs: print(l)
        await b.close()
asyncio.run(main())

File: tests/shots.mjs
/* Screenshots of the main states: node tests/shots.mjs [light|dark] -> tests/out/*.png */
import path from 'path';
import { serve, launch, attachLogs, loadSample, OUT } from './common.mjs';

const scheme = process.argv[2] || 'light';
const site = await serve();
const b = await launch();
// the sample lives in Lisbon: Records.json has no UTC offsets and falls back to the browser time zone (AGENTS.md, section 9)
const pg = await b.newPage({ viewport: { width: 1440, height: 900 }, colorScheme: scheme, timezoneId: 'Europe/Lisbon' });
const logs = [];
attachLogs(pg, logs);
const shot = name => pg.screenshot({ path: path.join(OUT, `${name}_${scheme}.png`) });
await pg.goto(site.url);
await pg.waitForTimeout(2500);
await shot('0_landing');
await loadSample(pg, site.url);
await shot('1_map');
await pg.click('#viewSeg button[data-v=split]'); await pg.waitForTimeout(5000);
await shot('2_split');
await pg.click('#colorSeg button[data-v=mode]');
await pg.evaluate("window.__itinera.setTimeRange(Date.UTC(2025, 8, 8), Date.UTC(2025, 8, 17), 'x')"); await pg.waitForTimeout(4000);
await shot('3_filtered');
await pg.evaluate("window.__itinera.setTimeRange(null, null, 'x'); window.__itinera.map.jumpTo({ center: [-9.14, 38.73], zoom: 11.5 })"); await pg.waitForTimeout(1500);
await pg.evaluate('window.__itinera.seek(Date.UTC(2025, 5, 4, 7, 0))');
await pg.click('#playBtn'); await pg.waitForTimeout(3000); await pg.click('#playBtn'); await pg.waitForTimeout(2500);
await shot('4_playback');
await pg.click('#viewSeg button[data-v=cube]'); await pg.waitForTimeout(4000);
await shot('5_cube');
// place timetable: all days stacked, then one week along the calendar
await pg.click('#stopBtn'); await pg.click('#colorSeg button[data-v=device]');
await pg.click('#viewSeg button[data-v=marey]'); await pg.waitForTimeout(2000);
await shot('7_timetable');
await pg.click('#mareySeg button[data-v=calendar]');
await pg.evaluate("window.__itinera.setTimeRange(Date.UTC(2025, 4, 5), Date.UTC(2025, 4, 12), 'x')"); await pg.waitForTimeout(2000);
await shot('8_timetable_week');
await pg.click('#mareySeg button[data-v=days]'); await pg.evaluate("window.__itinera.setTimeRange(null, null, 'x')");
await pg.click('#viewSeg button[data-v=split]'); await pg.waitForTimeout(3000);
await shot('9_split_timetable');
// country scale: the built-in outline map (coastlines and borders)
await pg.click('#viewSeg button[data-v=map]'); await pg.click('#stopBtn');
await pg.evaluate('window.__itinera.map.jumpTo({ center: [-3, 40], zoom: 4.2 })'); await pg.waitForTimeout(3000);
await shot('6_country');
for (const l of logs) console.log(l);
await b.close(); site.close();
